tsdb/agent/db.go

```diff
@@ -985,11 +985,32 @@ func (a *appender) UpdateMetadata(storage.SeriesRef, labels.Labels, metadata.Met
 
 // Commit submits the collected samples and purges the batch.
 func (a *appender) Commit() error {
+	if err := a.log(); err != nil {
+		return err
+	}
+
+	a.pendingSeries = a.pendingSeries[:0]
+	a.pendingSamples = a.pendingSamples[:0]
+	a.pendingHistograms = a.pendingHistograms[:0]
+	a.pendingFloatHistograms = a.pendingFloatHistograms[:0]
+	a.pendingExamplars = a.pendingExamplars[:0]
+	a.sampleSeries = a.sampleSeries[:0]
+	a.histogramSeries = a.histogramSeries[:0]
+	a.floatHistogramSeries = a.floatHistogramSeries[:0]
```

Comment:
Can you extract these 8 lines which are now repeated in `Rollback`?

Reply:
I took a few attempts at extracting the lines, but found the logic confusing, since Commit should clear out everything after calling log, but Rollback should clear out everything except pendingSeries _before_ calling log. 

I resolved this creating a new method which only logs pending series which is called from Rollback. That allowed both Commit and Rollback to have the same general flow and allowed me to extract these 8 lines.